anchore_engine/services/policy_engine/engine/vulns/mappers.py

```diff
@@ -317,6 +317,9 @@ def image_content_to_grype_sbom(self, record: Dict):
         """
         artifact = super().image_content_to_grype_sbom(record)
         artifact["language"] = self.grype_language
+        if self.grype_language == "java" and "metadata" in record:
```

Comment:
I recommend a new mapper for Java extending the CPE mapper. Those mappers are designed to be extensible to allow for different logic based on package type. Otherwise the checks for package types will be spread through out the system and would be prone to errors 

Some pseudo code for the suggestion 
- override image_content_to_grype_sbom() 
- update the java mapper in ENGINE_PACKAGE_MAPPERS and GRYPE_PACKAGE_MAPPERS 
- move the unit tests for transformation logic

```
class JavaMapper(CPEMapper):
    def __init__(self):
        super(JavaMapper, self).__init__(
            engine_type="java", grype_type="java-archive", grype_language="java"
        )

    @staticmethod
    def _ic_to_grype_metadata(metadata: Optional[str]):
        result = None
        if not metadata:
            return result
        # transformation logic here

        return result

    def image_content_to_grype_sbom(self, record: Dict):
        artifact = super().image_content_to_grype_sbom(record)
        grype_metadata = self._ic_to_grype_metadata(record.get("metadata"))
        if grype_metadata:
            artifact["metadataType"] = "JavaMetadata"
            artifact["metadata"] = grype_metadata
```